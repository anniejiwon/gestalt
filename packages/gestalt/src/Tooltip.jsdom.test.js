// @flow
import React from 'react';
import { create } from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import Tooltip from './Tooltip.js';

test('Tooltip renders', () => {
  const component = create(
    <Tooltip text="This is a tooltip">
      <div>Hi</div>
    </Tooltip>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tooltip renders the hrefText when hovered', () => {
  const { getByRole, getByText } = render(
    <Tooltip
      href="pinterest.com"
      hrefText="Learn more about Pinterest"
      text="This is a tooltip"
    >
      <div>Hi</div>
    </Tooltip>
  );

  fireEvent.mouseEnter(getByRole('tooltip'));
  expect(getByText('Learn more about Pinterest')).toBeVisible();
});

test('Tooltip should render as expected when hovered', () => {
  const { container, getByRole } = render(
    <Tooltip text="This is a tooltip">
      <div>Hi</div>
    </Tooltip>
  );

  fireEvent.mouseEnter(getByRole('tooltip'));
  expect(container.firstChild).toMatchSnapshot();
});
