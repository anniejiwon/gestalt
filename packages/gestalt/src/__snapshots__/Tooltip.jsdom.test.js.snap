// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders 1`] = `
<div
  className="box xsDisplayBlock"
>
  <div
    className="box"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="tooltip"
  >
    <div>
      Hi
    </div>
  </div>
</div>
`;

exports[`Tooltip should render as expected when hovered 1`] = `
<div
  class="box xsDisplayBlock"
>
  <div
    class="box"
    role="tooltip"
  >
    <div>
      Hi
    </div>
  </div>
  <div
    class="container"
    style="visibility: visible; top: -8px; left: -28px;"
  >
    <div
      class="darkGrayBg darkGray contents maxDimensions"
      tabindex="-1"
    >
      <div
        class="innerContents maxDimensions"
      >
        <div
          class="box paddingX2 paddingY1"
          style="max-width: 180px;"
          tabindex="0"
        >
          <div
            class="Text fontSize1 white alignLeft breakWord fontWeightNormal"
          >
            This is a tooltip
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
